dotnet/Cucumber.Messages.Specs/MessagesTest.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Io.Cucumber.Messages.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Cucumber.Messages.Specs
+{
+    public class MessagesTest
+    {
+
+        [Fact]
+        public void ThrowsWhenRequiredFieldsAreMissing()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Background( null, null, null, null, null, null));
+        }
+
+        [Fact]
+        public void IsValidWhenNoRequiredFieldsAreMissing()
+        {
+            // This should succeed as no constructor arguments to an Envelope are required.
+            var env = new Envelope(null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null);
```

Comment:
Should this have an assertion like no error is raised?

Reply:
I was duplicating the approach taken by the Java implementation. It demonstrates that no arguments are required to the main constructor of an Envelope.
The alternative here, IMO, would be to remove the test entirely.

Reply:
There is an `assertDoesNotThrow(() -> { some code })` in Java, but its kinda redundant in single line test case.